frontend/src/AddTagView.svelte

```diff
@@ -0,0 +1,211 @@
+    text-align: center;
+  }
+
+  form {
+    max-width: 50%;
+  }
+
+  .center {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    width: 100%;
+    height: 100%;
+    color: var(--main-fg-color);
+  }
+
+  button,
+  select,
+  option {
+    background-color: var(--main-bg-color);
+    color: inherit;
+    border: 1px solid;
+    border-color: inherit;
+    border-radius: 0;
+    padding-top: 0.5em;
+    padding-bottom: 0.5em;
+    padding-left: 1em;
+    padding-right: 1em;
+    margin-left: 0.5em;
+    margin-right: 0.5em;
+    font-size: inherit;
+    font-family: var(--font);
+    box-shadow: none;
+  }
+
+  option {
+    font-family: monospace;
+  }
+
+  input[type="file"] {
+    display: none;
+  }
+</style>
+
+<script>
+  import { selected, selectedResource } from "./stores.js";
+  import { onMount } from "svelte";
+  import LoadingText from "./LoadingText.svelte";
+  import { cleanOfrakType } from "./helpers";
+
+  export let modifierView, resourceNodeDataMap, dataPromise;
+  let errorMessage,
+    ofrakTagsPromise = new Promise(() => {}),
+    selectedTag;
+
+  function refreshResource() {
+    // Force tree view children refresh
+    resourceNodeDataMap[$selected].collapsed = false;
+    resourceNodeDataMap[$selected].childrenPromise =
+      $selectedResource.get_children();
+
+    // Force hex view refresh with colors
+    const originalSelected = $selected;
+    $selected = undefined;
+    $selected = originalSelected;
+  }
+
+  function chooseTag() {
+    if (selectedTag) {
+      modifierView = undefined;
+      $selectedResource.add_tag(selectedTag);
+    }
+  }
+
+  onMount(async () => {
+    await fetch(`/get_all_tags`).then((r) => {
+      r.json().then((ofrakTags) => {
+        ofrakTags.sort(function (a, b) {
+          if (cleanOfrakType(a) > cleanOfrakType(b)) {
+            return 1;
+          } else {
+            return -1;
+          }
+        });
+        ofrakTagsPromise = ofrakTags;
+      });
+    });
+  });
+</script>
+
+<div class="container">
+  <div class="inputs">
+    <p>Select tag to add to resource.</p>
+    New tag:
+    {#await ofrakTagsPromise}
+      <LoadingText />
+    {:then ofrakTags}
+      {#if ofrakTagsPromise && ofrakTagsPromise.length > 0}
+        <form on:submit|preventDefault="{chooseTag}">
+          <select
+            on:click|stopPropagation="{() => undefined}"
+            bind:value="{selectedTag}"
+          >
+            <option value="{null}">Select a tag to add</option>
+            {#each ofrakTags as ofrakTag}
+              <option value="{ofrakTag}">
+                {cleanOfrakType(ofrakTag)}
+              </option>
+            {/each}
+          </select>
+
+          <button
+            on:click|stopPropagation="{() => undefined}"
+            disabled="{!selectedTag}"
+            type="submit">Add</button
+          >
+        </form>
+      {:else}
+        Failed to get the list of OFRAK tags!
```

Comment:
I think the following is more descriptive for this path.

```suggestion
        No tags found!
```